File: ClientSide/UkrainianStyleRestaurant_2.0_another/app/src/main/java/com/example/ukrainianstylerestaurant/net/AuthApi.java
package com.example.ukrainianstylerestaurant.net;

import com.example.ukrainianstylerestaurant.model.ClientProfileResponse;
import com.example.ukrainianstylerestaurant.model.LoginRequest;
import com.example.ukrainianstylerestaurant.model.LoginResponse;
import com.example.ukrainianstylerestaurant.model.RegisterRequest;
import com.example.ukrainianstylerestaurant.model.SaveTokenRequest;
import com.example.ukrainianstylerestaurant.model.UpdateProfileRequest;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface AuthApi {

    @POST("api/client/login")
    Call<LoginResponse> login(@Body LoginRequest request);

    @POST("api/client/register")
    Call<ResponseBody> register(@Body RegisterRequest request);

    @PUT("api/client/profile")
    Call<ResponseBody> updateProfile(@Body UpdateProfileRequest request);

    @POST("api/notifications/token")
    Call<ResponseBody> saveToken(@Body SaveTokenRequest request);

    @GET("api/client/{id}")
    Call<ClientProfileResponse> getProfile(@Path("id") int id);
}

File: StuffSide/WaiterApp/app/src/main/java/com/waiter/app/data/api/AuthApi.kt
package com.waiter.app.data.api

import com.waiter.app.data.dto.LoginRequest
import com.waiter.app.data.dto.LoginResponse
import com.waiter.app.data.dto.RegisterRequest
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthApi {
    // ... Waiter methods ...
    @POST("api/waiter/login")
    suspend fun loginWaiter(@Body request: LoginRequest): Response<LoginResponse>

    @POST("api/waiter/register")
    suspend fun registerWaiter(@Body request: RegisterRequest): Response<Unit>

    // ... Courier methods ...
    @POST("api/courier/login")
    suspend fun loginCourier(@Body request: LoginRequest): Response<LoginResponse>

    @POST("api/courier/register")
    suspend fun registerCourier(@Body request: RegisterRequest): Response<Unit>

    // --- НОВІ МЕТОДИ КУХАРЯ ---
    @POST("api/cook/login")
    suspend fun loginCook(@Body request: LoginRequest): Response<LoginResponse>

    @POST("api/cook/register")
    suspend fun registerCook(@Body request: RegisterRequest): Response<Unit>
}